Client/mods/deathmatch/logic/CClientVehicle.cpp

```diff
@@ -1117,8 +1117,12 @@ void CClientVehicle::SetVariant(unsigned char ucVariant, unsigned char ucVariant
 
     // clear our component data to regenerate it
     m_ComponentData.clear();
-
     ReCreate();
+
+    if (!m_bEngineOn)
```

Comment:
What is the purpose of this condition?

Reply:
The only thing the SetEngineOn doing is toggling a flag inside CVehicle object. So there's no overhead in calling that function and thus this condition can be omitted.

Reply:
> if the engine is on then no need to call `m_pVehicle->SetEngineOn(m_bEngineOn);` since using setVehicleVariant will recreate the vehicle and will set the engine to true but why setVehicleVariant recreate the vehicle anyway?
> 
> ```
> void CClientVehicle::SetVariant(unsigned char ucVariant, unsigned char ucVariant2)
> {
>     m_ucVariation = ucVariant;
>     m_ucVariation2 = ucVariant2;
> 
>     // clear our component data to regenerate it
>     m_ComponentData.clear();
>     ReCreate();
> 
>     if (!m_bEngineOn)
>     {
>         m_pVehicle->SetEngineOn(m_bEngineOn); 
>     }
> }
> ```

it needs to recreate the vehicle because the vehicle model changes

Reply:
I will check later for the root of this problem 
tho Create()
set  m_pVehicle->SetEngineOn(m_bEngineOn);
so value should be false 
getVehicleEngineState still returns false tho the engine is ON 
which really strange will need few tests and checking

Reply:
pls, CClientVehicle::IsEngineOn and CClientVehicle::SetEngineOn

Reply:
```
    if (!IsEngineOn())
    {
        SetEngineOn(m_bEngineOn); 
    }
```

Reply:
Maybe you wanted to do something like that?
```
if (!m_bEngineOn)
    SetEngineOn(m_bEngineOn)
```

Reply:
the problem within this line of code set the engine after the warp 

```
 // Eventually warp driver back in
 if (m_pDriver)
     m_pDriver->WarpIntoVehicle(this, 0);

```